src/main/java/hu/elte/eserial/builder/MapBuilder.java

```diff
@@ -0,0 +1,106 @@
+package hu.elte.eserial.builder;
+
+import hu.elte.eserial.exception.EserialBuilderMismatchException;
+import hu.elte.eserial.exception.EserialInstantiationException;
+import hu.elte.eserial.exception.EserialInvalidMethodException;
+import hu.elte.eserial.util.TypeUtils;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentSkipListMap;
+
+/**
+ * Build Map objects.
+ */
+public class MapBuilder extends AbstractBuilder{
+
+    /**
+     * Constructs an {@link MapBuilder} and sets the {@code type} in it.
+     *
+     * @param type {@link Map} class to be used in the {@link AbstractBuilder#build} method
+     */
+    MapBuilder(Type type) {
+        super(type);
+    }
+
+    /**
+     * @param value {@inheritDoc}
+     * @param <T> {@inheritDoc}
+     * @return a map of the given class and initialized with the map parameter
+     */
+    @Override
+    public <T> T build(Object value) {
+        if (value == null) {
+            return null;
+        }
+
+        Class clazz = TypeUtils.convertTypeToClass(type);
+        Type keyTypeArg = TypeUtils.getTypeArgument(type, 0);
+        Type valueTypeArg = TypeUtils.getTypeArgument(type, 1);
+
+        if (!TypeUtils.isMap(clazz) || !TypeUtils.isMap(value.getClass())) {
+            throw new EserialBuilderMismatchException(Map.class.getSimpleName(), clazz.getName());
+        }
+
+        try {
+            Map<Object, Object> map = (Map<Object, Object>) value;
+
+            Class keyTypeArgClass = TypeUtils.convertTypeToClass(keyTypeArg);
+            Class valueTypeArgsClass = TypeUtils.convertTypeToClass(valueTypeArg);
+
+            if (keyTypeArg != null && valueTypeArg != null) {
+                Map<Object, Object> newMap = new HashMap<>();
+                Iterator it = map.entrySet().iterator();
+                while(it.hasNext()) {
+                    Map.Entry pair = (Map.Entry)it.next();
+
+                    Object keyObject;
```

Comment:
I don't think it's a good idea to support two separate procedures for map serialization and deserialization because it adds unnecessary complexity (even though it might reduce the size of the output) . The current serialization workflow of maps is generalized - why would we add a specialized version as well? It can be an optimization in the future but I think it's out of scope of this assignment as for now.

Reply:
I'll have to agree, we surely do have more important things to focus on right now.